.infrastructure/helm-chart/todoapp/charts/mysql/templates/statefulSet.yml

```diff
@@ -0,0 +1,89 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.namespace}}
+spec:
+  selector:
+    matchLabels:
+      app: {{.Chart.Name}}
+  replicas: {{.Values.mysql.statefulSet.replicas}}
+  serviceName: {{.Chart.Name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Chart.Name}}
+    spec:
+      containers:
+      - name: {{.Chart.Name}}
+        image: {{.Values.mysql.image.repository}}:{{.Values.mysql.image.tag}}
+        env:
+        {{- range $k, $v := .Values.mysql.secrets}}
+        - name: {{$k}}
+          valueFrom:
+            secretKeyRef:
+              name: mysql-secrets
+              key: {{$k}}
+        {{- end}}
+        - name: MYSQL_DATABASE
+          value: app_db
+        ports:
+        - name: {{.Chart.Name}}
+          containerPort: 3306
+        volumeMounts:
+        - name: data
+          mountPath: /var/lib/mysql
+          subPath: {{.Chart.Name}}
+        - name: config-map
+          mountPath: /docker-entrypoint-initdb.d
+        resources:
+          requests:
+            cpu: {{.Values.mysql.resources.cpu.request}}
+            memory: {{.Values.mysql.resources.memory.request}}
+        livenessProbe:
+          exec:
+            command: ["mysqladmin", "ping"]
+          initialDelaySeconds: 30
+          periodSeconds: 10
+          timeoutSeconds: 5
+        readinessProbe:
+          exec:
+            command: ["mysqladmin", "ping"]
+          initialDelaySeconds: 5
+          periodSeconds: 2
+          timeoutSeconds: 1
+      volumes:
+      - name: config-map
+        configMap:
+          name: {{.Chart.Name}}
+      tolerations:
+      - key: {{ .Values.mysql.toleration.key}}
+        operator: {{ .Values.mysql.toleration.operator}}
+        value: {{ .Values.mysql.toleration.value}}
+        effect: {{ .Values.mysql.toleration.effect}}
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: {{ .Values.mysql.podAntiAffinity.key}}
+                    operator: {{ .Values.mysql.podAntiAffinity.operator}}
+                    values:
+                    {{ .Values.mysql.podAntiAffinity.values}}
+              topologyKey: {{ .Values.mysql.podAntiAffinity.topologyKey}}
```

Comment:
Ensure that the `mysql.podAntiAffinity` values (key, operator, values, topologyKey) are defined in your `values.yaml` file. If they are not defined, this could lead to issues when deploying the chart. Consider providing default values or handling the case where they might be undefined.